usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateAllOptionalFileName.java

```diff
@@ -0,0 +1,40 @@
+package org.mobilitydata.gtfsvalidator.usecase;
+
+import org.mobilitydata.gtfsvalidator.usecase.notice.OptionalFileFoundNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.GtfsSpecRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.RawFileRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+import java.util.List;
+
+public class ValidateAllOptionalFileName {
+
+    private final GtfsSpecRepository specRepo;
+    private final RawFileRepository rawFileRepo;
+    private final ValidationResultRepository resultRepo;
+    private final List<String> filenameList;
+
+    public ValidateAllOptionalFileName(final GtfsSpecRepository specRepo, final RawFileRepository rawFileRepo,
+                                       final ValidationResultRepository resultRepo, List<String> filenameList) {
+        this.specRepo = specRepo;
+        this.rawFileRepo = rawFileRepo;
+        this.resultRepo = resultRepo;
+        this.filenameList = filenameList;
+    }
+
+    public List<String> execute() {
+
+        specRepo.getOptionalFilenameList().stream()
+                .filter(optionalFilename -> rawFileRepo.getFilenameAll().contains(optionalFilename))
+                .forEach(optionalFilename -> {
+                            filenameList.add(optionalFilename);
```

Comment:
Why are we adding INFO notices for each optional file? Just curious - it's not something I would have thought to do. Are you using these notices later for something?

Reply:
Good remark. I realize that the best option should be to add an Info notice for each missing optional file. Having this information, the person using the validator would be aware of any missing optional file he/she tried to validate. 

It might be a bit memory consuming, but at the same time we would generate at most 10 notices.

Maybe this is not something that we'd like .. we can discuss this with @fabrice-v  and @LeoFrachet

Reply:
The use for info notices is not very defined yet. The middle/long term idea is to have an optional verbosity option that would surface them.